### ISSUE TITLE
Bad memory usage in VisualShaderNodeGroupBase

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#32722]
**Godot version:**
3.2.alpha.custom_build. 4ee8ecd3e

**OS/device including version:**
Ubuntu 18.04

**Issue description:**
ASAN report
```
==23210==ERROR: AddressSanitizer: heap-use-after-free on address 0x6190000e3390 at pc 0x00000e5fd666 bp 0x7ffe84bd50b0 sp 0x7ffe84bd50a0
READ of size 8 at 0x6190000e3390 thread T0
    #0 0xe5fd665 in Object::to_string() core/object.cpp:939
    #1 0xe886f62 in Variant::stringify(List<void const*, DefaultAllocator>&) const core/variant.cpp:1592
    #2 0xe881f8a in Variant::operator String() const core/variant.cpp:1408
    #3 0x1a8ba3e in GDScriptFunctions::call(GDScriptFunctions::Function, Variant const**, int, Variant&, Variant::CallError&) modules/gdscript/gdscript_functions.cpp:701
    #4 0x1a1951c in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_function.cpp:1146
    #5 0x184f376 in GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) modules/gdscript/gdscript.cpp:1180
    #6 0x9377244 in Node::_notification(int) scene/main/node.cpp:60
    #7 0x17088b7 in Node::_notificationv(int, bool) scene/main/node.h:46
    #8 0x170ad09 in CanvasItem::_notificationv(int, bool) scene/2d/canvas_item.h:166
    #9 0xadabc51 in Node2D::_notificationv(int, bool) scene/2d/node_2d.h:38
    #10 0xe5fcea9 in Object::notification(int, bool) core/object.cpp:931
    #11 0x94a743d in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:964
    #12 0x9497fd6 in SceneTree::idle(float) scene/main/scene_tree.cpp:516
    #13 0x15851c2 in Main::iteration() main/main.cpp:1978
    #14 0x14827a2 in OS_X11::run() platform/x11/os_x11.cpp:3192
    #15 0x13feba3 in main platform/x11/godot_x11.cpp:56
    #16 0x7f51b002db6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)
    #17 0x13fe7a9 in _start (/usr/bin/godots+0x13fe7a9)

0x6190000e3390 is located 16 bytes inside of 1104-byte region [0x6190000e3380,0x6190000e37d0)
freed by thread T0 here:
    #0 0x7f51b192904f in __interceptor_free (/lib/x86_64-linux-gnu/libasan.so.5+0x10c04f)
    #1 0xeb9a29b in Memory::free_static(void*, bool) core/os/memory.cpp:181
    #2 0x158fa38 in void memdelete<Object>(Object*) core/os/memory.h:119
    #3 0x94ab364 in SceneTree::_flush_delete_queue() scene/main/scene_tree.cpp:1110
    #4 0x94995ab in SceneTree::idle(float) scene/main/scene_tree.cpp:534
    #5 0x15851c2 in Main::iteration() main/main.cpp:1978
    #6 0x14827a2 in OS_X11::run() platform/x11/os_x11.cpp:3192
    #7 0x13feba3 in main platform/x11/godot_x11.cpp:56
    #8 0x7f51b002db6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)

previously allocated by thread T0 here:
    #0 0x7f51b1929448 in malloc (/lib/x86_64-linux-gnu/libasan.so.5+0x10c448)
    #1 0xeb99246 in Memory::alloc_static(unsigned long, bool) core/os/memory.cpp:85
    #2 0xeb99142 in operator new(unsigned long, char const*) core/os/memory.cpp:42
    #3 0x92e074c in Object* ClassDB::creator<Control>() core/class_db.h:142
    #4 0xe3889a0 in ClassDB::instance(StringName const&) core/class_db.cpp:541
    #5 0x1828dc0 in GDScriptNativeClass::instance() modules/gdscript/gdscript.cpp:82
    #6 0x1828669 in GDScriptNativeClass::_new() modules/gdscript/gdscript.cpp:69
    #7 0x18f906f in MethodBind0R<Variant>::call(Object*, Variant const**, int, Variant::CallError&) core/method_bind.gen.inc:237
    #8 0xe5fc93e in Object::call(StringName const&, Variant const**, int, Variant::CallError&) core/object.cpp:921
    #9 0xe8eb494 in Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) core/variant_call.cpp:1083
    #10 0x1a177b9 in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_function.cpp:1079
    #11 0x184f376 in GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) modules/gdscript/gdscript.cpp:1180
    #12 0x9377244 in Node::_notification(int) scene/main/node.cpp:60
    #13 0x17088b7 in Node::_notificationv(int, bool) scene/main/node.h:46
    #14 0x170ad09 in CanvasItem::_notificationv(int, bool) scene/2d/canvas_item.h:166
    #15 0xadabc51 in Node2D::_notificationv(int, bool) scene/2d/node_2d.h:38
    #16 0xe5fcea9 in Object::notification(int, bool) core/object.cpp:931
    #17 0x94a743d in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:964
    #18 0x9497fd6 in SceneTree::idle(float) scene/main/scene_tree.cpp:516
    #19 0x15851c2 in Main::iteration() main/main.cpp:1978
    #20 0x14827a2 in OS_X11::run() platform/x11/os_x11.cpp:3192
    #21 0x13feba3 in main platform/x11/godot_x11.cpp:56
    #22 0x7f51b002db6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)

```

**Minimal reproduction project:**
[Visual.zip](https://github.com/godotengine/godot/files/3713945/Visual.zip)

**Comment by Chaosus** [comment]
> Bad memory usage in

Do you mean its a memory leak ?

**Comment by qarmin** [comment]
Reading from freed memory

**Comment by Chaosus** [comment]
> Reading from freed memory

Hmm, then it should lead to the crash. But I didn't detect crash on aad4d8648b4593b5edb479e85984730770e77c70 (Windows10)

**Comment by qarmin** [comment]
Fixed in Godot 3.2.2, because broken functions were deleted/hided
